### ISSUE TITLE
Multiple metrics aggregation throwing an error

### ISSUE THREAD

**Issue by mouhamadalmounayar** [Green-Software-Foundation/if#1035]
## Description of the Error

IF fails to run a manifest when multiple aggregation metrics are specified. According to the online documentation, it should be possible to pass an array of metrics for aggregation, but this is resulting in an error.

## Expected Behaviour

We should get an output manifest with both energy and carbon metrics aggregated. 
## Actual Behaviour

The following error is thrown : 
```
[2024-10-03 03:46:00.471 PM] error:     Aggregation metric carbon is not found in inputs[0].
MissingAggregationParamError:   Aggregation metric carbon is not found in inputs[0].
    at C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\util\aggregation-helper.js:20:23
    at Array.reduce (<anonymous>)
    at aggregateInputsIntoOne (C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\util\aggregation-helper.js:17:19)
    at aggregateNode (C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\lib\aggregate.js:62:79)
    at aggregateNode (C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\lib\aggregate.js:50:13)
    at aggregate (C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\lib\aggregate.js:76:5)
    at impactEngine (C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\index.js:45:58)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```
## Steps to Reproduce

Run the following manifest file in order to reproduce the bug.

## Manifest File That Generated the Error
```yaml
name: carbon_pipeline
aggregation:
  metrics:
  - carbon
  - energy
  type: vertical
initialize:
  plugins:
    teads-curve:
      method: TeadsCurve
      path: '@grnsft/if-unofficial-plugins'
    e-mem:
      method: EMem
      path: '@grnsft/if-plugins'
    sci-e:
      method: SciE
      path: '@grnsft/if-plugins'
    sci-m:
      method: SciM
      path: '@grnsft/if-plugins'
    sci-o:
      method: SciO
      path: '@grnsft/if-plugins'
    sci:
      method: Sci
      path: '@grnsft/if-plugins'
  outputs:
  - yaml
tree:
  defaults:
    memory/utilization: 100
    device/emissions-embodied: 1887000
    cpu/thermal-design-power: 4.66
    device/expected-lifespan: 126230400
    duration: 1800
  pipeline:
    compute:
    - teads-curve
    - e-mem
    - sci-e
    - sci-m
    - sci-o
    - sci
  config:
    sci:
      functional-unit: R
  children:
    0:
      children:
        child_1:
          inputs:
          - timestamp: 1727955000
            cpu/utilization: 36.437729148146225
            grid/carbon-intensity: 251.0
            resources-reserved: 1.0
            resources-total: 90
            memory/capacity: 2.576980378
          - timestamp: 1727956800
            cpu/utilization: 35.54560485752586
            grid/carbon-intensity: 251.0
            resources-reserved: 1.0
            resources-total: 90
            memory/capacity: 2.576980378
          - timestamp: 1727958600
            cpu/utilization: 35.32553725926037
            grid/carbon-intensity: 251.0
            resources-reserved: 1.0
            resources-total: 90
            memory/capacity: 2.576980378
          - timestamp: 1727960400
            cpu/utilization: 33.50961455555754
            grid/carbon-intensity: 251.0
            resources-reserved: 1.0
            resources-total: 90
            memory/capacity: 2.576980378
          - timestamp: 1727962200
            cpu/utilization: 34.59113314814911
            grid/carbon-intensity: 251.0
            resources-reserved: 1.0
            resources-total: 90
            memory/capacity: 2.576980378
          - timestamp: 1727964000
            cpu/utilization: 38.092500888888026
            grid/carbon-intensity: 251.0
            resources-reserved: 1.0
            resources-total: 90
            memory/capacity: 2.576980378
```
This manifest works fine when only one metric is specified, and generates the following output : 

```yaml
name: carbon_pipeline
aggregation:
  metrics:
    - carbon
  type: vertical
initialize:
  plugins:
    teads-curve:
      path: '@grnsft/if-unofficial-plugins'
      method: TeadsCurve
    e-mem:
      path: '@grnsft/if-plugins'
      method: EMem
    sci-e:
      path: '@grnsft/if-plugins'
      method: SciE
    sci-m:
      path: '@grnsft/if-plugins'
      method: SciM
    sci-o:
      path: '@grnsft/if-plugins'
      method: SciO
    sci:
      path: '@grnsft/if-plugins'
      method: Sci
execution:
  command: >-
    C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node.exe
    C:\Users\malmounayar\programs\node-v20.17.0-win-x64\node_modules\@grnsft\if\build\if-run\index.js
    --manifest
    C:\Users\malmounayar\GreenIt\carbon-engine\backend\src\services\carbon_service\impact_framework\files\generated\if_input.yaml
    --output
    C:\Users\malmounayar\GreenIt\carbon-engine\backend\src\services\carbon_service\impact_framework\files\generated\if_output
  environment:
    if-version: 0.6.0
    os: "Microsoft Windows 11 Enterprise\r"
    os-version: 10.0.22631 N/A Build 22631
    node-version: 20.17.0
    date-time: 2024-10-03T14:30:27.254Z (UTC)
    dependencies:
      - '@grnsft/if-plugins@0.3.2'
      - '@grnsft/if-unofficial-plugins@0.3.1'
      - '@grnsft/if@0.6.0'
      - ts-node@10.9.1
  status: success
tree:
  defaults:
    memory/utilization: 100
    device/emissions-embodied: 1887000
    cpu/thermal-design-power: 4.66
    device/expected-lifespan: 126230400
    duration: 1800
  pipeline:
    compute:
      - teads-curve
      - e-mem
      - sci-e
      - sci-m
      - sci-o
      - sci
  config:
    sci:
      functional-unit: R
  children:
    '0':
      children:
        child_1:
          inputs:
            - timestamp: 1727958600
              cpu/utilization: 35.32553725926037
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
            - timestamp: 1727960400
              cpu/utilization: 33.50961455555754
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
            - timestamp: 1727962200
              cpu/utilization: 34.59113314814911
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
            - timestamp: 1727964000
              cpu/utilization: 36.707015481480525
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
            - timestamp: 1727965800
              cpu/utilization: 37.012663296296125
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
            - timestamp: 1727967600
              cpu/utilization: 37.012663296296125
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
          outputs:
            - timestamp: 1727958600
              cpu/utilization: 35.32553725926037
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
              memory/utilization: 100
              device/emissions-embodied: 1887000
              cpu/thermal-design-power: 4.66
              device/expected-lifespan: 126230400
              duration: 1800
              cpu/energy: 0.0014988881221778647
              memory/energy: 0.001010176308176
              energy: 0.0025090644303538646
              carbon-embodied: 0.2989771085253632
              carbon-operational: 0.62977517201882
              carbon: 0.0005159734891912129
              sci: 0.0005159734891912129
            - timestamp: 1727960400
              cpu/utilization: 33.50961455555754
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
              memory/utilization: 100
              device/emissions-embodied: 1887000
              cpu/thermal-design-power: 4.66
              device/expected-lifespan: 126230400
              duration: 1800
              cpu/energy: 0.0014617544632226037
              memory/energy: 0.001010176308176
              energy: 0.0024719307713986036
              carbon-embodied: 0.2989771085253632
              carbon-operational: 0.6204546236210495
              carbon: 0.0005107954067480071
              sci: 0.0005107954067480071
            - timestamp: 1727962200
              cpu/utilization: 34.59113314814911
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
              memory/utilization: 100
              device/emissions-embodied: 1887000
              cpu/thermal-design-power: 4.66
              device/expected-lifespan: 126230400
              duration: 1800
              cpu/energy: 0.0014840951764785947
              memory/energy: 0.001010176308176
              energy: 0.0024942714846545944
              carbon-embodied: 0.2989771085253632
              carbon-operational: 0.6260621426483032
              carbon: 0.0005139106950964814
              sci: 0.0005139106950964814
            - timestamp: 1727964000
              cpu/utilization: 36.707015481480525
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
              memory/utilization: 100
              device/emissions-embodied: 1887000
              cpu/thermal-design-power: 4.66
              device/expected-lifespan: 126230400
              duration: 1800
              cpu/energy: 0.0015259302403253824
              memory/energy: 0.001010176308176
              energy: 0.002536106548501382
              carbon-embodied: 0.2989771085253632
              carbon-operational: 0.6365627436738469
              carbon: 0.0005197443623328945
              sci: 0.0005197443623328945
            - timestamp: 1727965800
              cpu/utilization: 37.012663296296125
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
              memory/utilization: 100
              device/emissions-embodied: 1887000
              cpu/thermal-design-power: 4.66
              device/expected-lifespan: 126230400
              duration: 1800
              cpu/energy: 0.0015317804967879887
              memory/energy: 0.001010176308176
              energy: 0.0025419568049639884
              carbon-embodied: 0.2989771085253632
              carbon-operational: 0.6380311580459611
              carbon: 0.0005205601480951802
              sci: 0.0005205601480951802
            - timestamp: 1727967600
              cpu/utilization: 37.012663296296125
              grid/carbon-intensity: 251
              resources-reserved: 1
              resources-total: 90
              memory/capacity: 2.576980378
              memory/utilization: 100
              device/emissions-embodied: 1887000
              cpu/thermal-design-power: 4.66
              device/expected-lifespan: 126230400
              duration: 1800
              cpu/energy: 0.0015317804967879887
              memory/energy: 0.001010176308176
              energy: 0.0025419568049639884
              carbon-embodied: 0.2989771085253632
              carbon-operational: 0.6380311580459611
              carbon: 0.0005205601480951802
              sci: 0.0005205601480951802
      outputs:
        - carbon: 0.0005159734891912129
          timestamp: 1727958600
          duration: 1800
        - carbon: 0.0005107954067480071
          timestamp: 1727960400
          duration: 1800
        - carbon: 0.0005139106950964814
          timestamp: 1727962200
          duration: 1800
        - carbon: 0.0005197443623328945
          timestamp: 1727964000
          duration: 1800
        - carbon: 0.0005205601480951802
          timestamp: 1727965800
          duration: 1800
        - carbon: 0.0005205601480951802
          timestamp: 1727967600
          duration: 1800
      aggregated:
        carbon: 0.0031015442495589563
  outputs:
    - carbon: 0.0005159734891912129
      timestamp: 1727958600
      duration: 1800
    - carbon: 0.0005107954067480071
      timestamp: 1727960400
      duration: 1800
    - carbon: 0.0005139106950964814
      timestamp: 1727962200
      duration: 1800
    - carbon: 0.0005197443623328945
      timestamp: 1727964000
      duration: 1800
    - carbon: 0.0005205601480951802
      timestamp: 1727965800
      duration: 1800
    - carbon: 0.0005205601480951802
      timestamp: 1727967600
      duration: 1800
  aggregated:
    carbon: 0.0031015442495589563
```

**Comment by jmcook1186** [comment]
Hi @mouhamadalmounayar - yes, i can repro your error using the IF npm package; however, it is fixed in the latest development version. We are going to publish a new release imminently (by tomorrow) and your issue should be magically fixed...!

**Comment by mouhamadalmounayar** [comment]
Hi @jmcook1186, is it normal that `if-run` is not working for the new release v0.7.0. It seems that the package is missing the `if-run/index.js` file, there is no `./build` directory like the previous versions.

**Comment by mouhamadalmounayar** [comment]
fixed with `0.7.1`
